internal/ext/stringsx/stringsx.go

```diff
@@ -17,12 +17,37 @@ package stringsx
 
 import (
 	"strings"
+	"unicode/utf8"
 
 	"github.com/bufbuild/protocompile/internal/ext/iterx"
+	"github.com/bufbuild/protocompile/internal/ext/slicesx"
 	"github.com/bufbuild/protocompile/internal/ext/unsafex"
 	"github.com/bufbuild/protocompile/internal/iter"
 )
 
+// Rune returns the rune at the given index.
```

Comment:
This is rather confusing. If you `range` over a string that has multi-byte runes, that's not what this index would be (despite it sounding from this doc like it could be).

Maybe clarify that it returns the rune that begins at this _byte index_.